### DISCUSSION TITLE
Build failure due to missing reference

### DISCUSSION THREAD

**Discussion by Genzord111** [idaholab/moose#28476] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

I very recently started exploring MOOSE. My first build keeps failing with the output below;

Linking Executable /home/max/projects/Al-Mg-Si_evo/evolution-opt...
/home/max/.conda/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/max/projects/Al-Mg-Si_evo/lib/libevolution-opt.so: undefined reference to `InputParameters validParams<AuxKernelTempl<double> >()'
collect2: error: ld returned 1 exit status
make: *** [/home/max/projects/moose/framework/app.mk:457: /home/max/projects/Al-Mg-Si_evo/evolution-opt] Error 1

It is a rerun of a simulation setup I found online at 
https://github.com/Sachinscnpdl/Al-Cu-Ni_dft_pf/tree/main/Phase_fields/Moose_codes.

Can I get some help with this?

**Comment by GiudGiud** [comment]
Hello

This code is deprecated, we no longer use templates to define validParams
See this announcement for how to transition your code
https://mooseframework.inl.gov/newsletter/2020/2020_04.html

Guillaume

**Comment by Genzord111** [comment]
Thank you, I have been able to resolve that issue. I'm trying to run one of
my problem inputs but
I get this error
*** ERROR ***
A 'SwitchingFunctionMultiPhaseMaterial' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make
sure that
the library can be found either in your "Problem/library_path" parameter or
in the
MOOSE_LIBRARY_PATH environment variable.

Could you help me with this?

Solomon

**Reply by GiudGiud** [reply]
did you activate the PhaseField module in the Makefile of your application?
If not, then you need to, then git commit that change, then git clean everything, then rebuild

**Comment by Genzord111** [comment]
I'm not so sure how to activate the PhaseField module in the makefile.

**Reply by GiudGiud** [reply]
if you open the Makefile in a text editor you can just write yes in front of PHASE_FIELD

**Comment by Genzord111** [comment]
I'm still getting the same error message please.

*** ERROR ***
A 'SwitchingFunctionMultiPhaseMaterial' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make
sure that
the library can be found either in your "Problem/library_path" parameter or
in the
MOOSE_LIBRARY_PATH environment variable.

Stack frames: 21
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char,
std::char_traits<char>, std::allocator<char> >,
std::__cxx11::basic_string<char, std::char_traits<char>,
std::allocator<char> >)
2: void mooseError<std::__cxx11::basic_string<char, std::char_traits<char>,
std::allocator<char> > >(std::__cxx11::basic_string<char,
std::char_traits<char>, std::allocator<char> >&&)
3: Factory::reportUnregisteredError(std::__cxx11::basic_string<char,
std::char_traits<char>, std::allocator<char> > const&) const
4: Factory::getValidParams(std::__cxx11::basic_string<char,
std::char_traits<char>, std::allocator<char> > const&) const
5: MooseObjectAction::MooseObjectAction(InputParameters const&)
6: AddMaterialAction::AddMaterialAction(InputParameters const&)
7: RegistryEntry<AddMaterialAction>::buildAction(InputParameters const&)
8: ActionFactory::create(std::__cxx11::basic_string<char,
std::char_traits<char>, std::allocator<char> > const&,
std::__cxx11::basic_string<char, std::char_traits<char>,
std::allocator<char> > const&, InputParameters&)
9: Moose::Builder::walkRaw(std::__cxx11::basic_string<char,
std::char_traits<char>, std::allocator<char> >,
std::__cxx11::basic_string<char, std::char_traits<char>,
std::allocator<char> >, hit::Node*)
10: Moose::Builder::walk(std::__cxx11::basic_string<char,
std::char_traits<char>, std::allocator<char> > const&,
std::__cxx11::basic_string<char, std::char_traits<char>,
std::allocator<char> > const&, hit::Node*)
11: hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder)
12: hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder)
13: hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder)
14: Moose::Builder::build()
15: MooseApp::setupOptions()
16: MooseApp::run()
17: ./evolution-opt(+0x20e8) [0x55a29a1530e8]
18: main
19: __libc_start_main
20: ./evolution-opt(+0x238b) [0x55a29a15338b]
Abort(1) on node 0 (rank 0 in comm 0): application called
MPI_Abort(MPI_COMM_WORLD, 1) - process 0

**Comment by Genzord111** [comment]
I have attached both my input file and Makefile

**Reply by GiudGiud** [reply]
I dont see them. You have to go on the GH website to reply with attachments.

**Comment by Genzord111** [comment]
Sorry, Pls find them attached. 

I should mention that I am practicing PhaseField simulations using an input file I forked. So I'm not really sure what I am doing.

[Makefile.txt](https://github.com/user-attachments/files/16767073/Makefile.txt)
[strain_0.0..txt](https://github.com/user-attachments/files/16767074/strain_0.0.txt)

**Reply by GiudGiud** [reply]
```
PHASE_FIELD                 := yes
```
this is correct.
now clean both repos and rebuild
you ll need to make sure to save those files you are working on as `git clean -Xfd` will remove them

**Comment by Genzord111** [comment]
Thank you very much for the help. I've been able to run my the code after
rebuilding my moose app.
